maven-repository-metadata/src/main/mdo/metadata.mdo

```diff
@@ -190,17 +192,24 @@ under the License.
                     {
                         s.setTimestamp( snapshot.getTimestamp() );
                         changed = true;
+                        updateSnapshotVersions = true;
                     }
                     if ( s.getBuildNumber() != snapshot.getBuildNumber() )
                     {
                         s.setBuildNumber( snapshot.getBuildNumber() );
                         changed = true;
+                        updateSnapshotVersions = true;
                     }
                     if ( s.isLocalCopy() != snapshot.isLocalCopy() )
                     {
                         s.setLocalCopy( snapshot.isLocalCopy() );
                         changed = true;
                     }
+                    if (updateSnapshotVersions)
+                    {
+                      v.setSnapshotVersions( new java.util.ArrayList<SnapshotVersion>( versioning.getSnapshotVersions() ) );
```

Comment:
I think this one is wrong because it has to work like versions. Test whether the collection contains if not, add it. You are replacing the list and not merging.